narwhal/network/src/admin.rs

```diff
@@ -48,7 +50,22 @@ pub fn start_admin_server(
 
     handles.push(spawn_logged_monitored_task!(
         async move {
-            axum_server::bind(socket_address)
+            let mut i = 0;
+            let listener = loop {
+                i += 1;
+                match TcpListener::bind(socket_address) {
+                    Ok(listener) => break listener,
+                    Err(e) => {
+                        if i == 10 {
+                            panic!("Failed to bind to {socket_address}: {e}");
+                        } else {
+                            warn!("Failed to bind to {socket_address}: {e}. Retrying ...");
+                            sleep(Duration::from_secs(1)).await;
+                        }
+                    }
+                };
+            };
+            axum_server::from_tcp(listener)
                 .handle(shutdown_handle)
                 .serve(router.into_make_service())
                 .await
```

Comment:
IIUC we shouldn't need to do this if we do the following:
1. send the shutdown signal to the server
2. await on the server to finish shutting down. I take it that we aren't properly waiting for the axum server to cleanly exit before we're trying to restart the server.

Reply:
How would you await on server to finish shutdown?

Reply:
>How would you await on server to finish shutdown?

This is what line 54 is doing, awaiting on the server. When the server shutsdown this will return

Reply:
I do see the `Future narwhal/network/src/admin.rs:AdminServerTask completed` log message. There should not be a race between joining on the admin server handler and starting a new admin server. If after server returns from await, the "real" socket binding must have dropped, then the "duplicate socket" error from simulator could be due to something else.